A/ADOLC/build_tarballs.jl

```diff
@@ -0,0 +1,73 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+# See https://discourse.julialang.org/t/binarybuilder-jl-cant-dlopen-because-of-libopenlibm-so/108486
+# This build script needs to be run with julia 1.7
+
+# See https://github.com/JuliaLang/Pkg.jl/issues/2942
+# Once this Pkg issue is resolved, this must be removed
+uuid = Base.UUID("a83860b7-747b-57cf-bf1f-3e79990d037f")
+delete!(Pkg.Types.get_last_stdlibs(v"1.6.3"), uuid)
+
+name = "ADOLC"
+version = v"0.0.24"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/coin-or/ADOL-C.git", "b7ea10536a1e452f7dd47a90ef8c8118ce3e8432"),
+    GitSource("https://github.com/TimSiebert1/libadolccxx.git", "51c23abf9ee2e1566331b440e81503ac9e7e8f5b"),
+]
+
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/ADOL-C/
+autoreconf -fi
+./configure --enable-atrig-erf --prefix=${prefix} --build=${MACHTYPE} --host=${target}
+make -j${nproc}
+make install
+install_license LICENSE
+
+cd ../libadolccxx/
+cmake -DCMAKE_INSTALL_PREFIX=${prefix}\
+      -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}\
+      -DCMAKE_BUILD_TYPE=Release\
+      -DADOLC_DIR=${prefix}\
+      -DJulia_PREFIX=${prefix}\
+      -B build .
+
+cd build
+make -j${nprocs}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+# See https://discourse.julialang.org/t/binarybuilder-jl-cant-dlopen-because-of-libopenlibm-so/108486
+# It seems we need a separate build for each Julia version
+include("../../L/libjulia/common.jl")
+
+julia_versions=VersionNumber[v"1.9", v"1.10", v"1.11"]
+
+platforms = vcat(libjulia_platforms.(julia_versions)...)
+
+# Current restriction:
+# ADOLC does not compile with musl libc.
+# Typical Error:
+# /opt/aarch64-linux-musl/aarch64-linux-musl/include/c++/7.1.0/cstdlib:144:11: error: ‘::calloc’ has not been declared 
+# Same for realloc and malloc.
+filter!(p->  (libc(p)!="musl"), platforms)
+
+products = [
+    LibraryProduct("libadolc_wrap", :libadolc_wrap)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    BuildDependency(PackageSpec(name="libcxxwrap_julia_jll", uuid="3eaa8342-bff7-56a5-9981-c04077f7cee7"))
```

Comment:
GitHub UI isn't exactly clear, but https://github.com/JuliaPackaging/Yggdrasil/pull/8872#discussion_r1638258789 was only on the `libjulia_jll` line, `libcxxwrap_julia_jll` is still needed at runtime.  Also, you do need to specify the compat for this package, since it's a runtime package with frequent breaking changes.

Reply:
To be crystal clear, https://github.com/JuliaPackaging/Yggdrasil/pull/8872#discussion_r1638258789 was referring to the line
```
    Dependency(PackageSpec(name="libjulia_jll", uuid="5ad3ddd2-0711-543a-b040-befd59781bbf");compat="=1.10.10"),
```
being a runtime dependency (libjulia should be only build-time) and forcing ***only*** v1.10.10.  I said nothing about `libcxxwrap_julia_jll`

Reply:
Ok.